pinecone/data/grpc/index_grpc.py

```diff
@@ -1,22 +1,19 @@
 import logging
 import numbers
-from abc import ABC, abstractmethod
-from functools import wraps
-from importlib.util import find_spec
-from typing import NamedTuple, Optional, Dict, Iterable, Union, List, Tuple, Any
+from typing import Optional, Dict, Iterable, Union, List, Tuple, Any
```

Comment:
Extracted a bunch of supporting classes and functions into smaller files. Removed dependencies no longer in use by the class defined in this file. Otherwise, no changes.